sthgetter/sthgetter_test.go

```diff
@@ -0,0 +1,187 @@
+	return uint64(t.UnixNano() / time.Millisecond.Nanoseconds())
+}
+
+func mustB64Decode(b64 string) []byte {
+	b, err := base64.StdEncoding.DecodeString(b64)
+	if err != nil {
+		panic(fmt.Sprintf("error decoding %s: %s", b64, err))
+	}
+	return b
+}
+
+var (
+	// A valid STH from the Google Pilot Log.
+	validSTH = &ct.GetSTHResponse{
+		TreeSize:          580682455,
+		Timestamp:         1554897886201, // 2019-04-10 12:04:46.201 UTC
+		SHA256RootHash:    mustB64Decode("VicMkhzrGNv+lNCwXRVHH0WniZuDg3IXhgPai5kyHdA="),
+		TreeHeadSignature: mustB64Decode("BAMARzBFAiEAs0GiYnPT5ZQJ2LGLhLmIXZXSLg+N+CxTkJL75tECEqgCIBZzJGyzH9h+IL63XCvRlfTKhLvzSxVicrT30+rwTSU0"),
+	}
+	// The public key for the Google Pilot Log.
+	b64PubKey = "MFkwEwYHKoZIzj0CAQYIKoZIzj0DAQcDQgAEfahLEimAoz2t01p3uMziiLOl/fHTDM0YDOhBRuiBARsV4UvxG2LdNgoIGLrtCzWE0J5APC2em4JlvR8EEEFMoA=="
+	// A time less than 24 hours after the timestamp of validSTH.  If validSTH
+	// was received at this time, it would not be considered 'too old'.
+	validReceiveTime   = time.Date(2019, time.April, 10, 15, 0, 0, 0, time.UTC)
+	invalidReceiveTime = time.Date(2019, time.April, 11, 15, 0, 0, 0, time.UTC)
+)
+
+func TestCheckSTH(t *testing.T) {
+	// Create Log structure.
+	l, err := ctlog.New("https://ct.googleapis.com", "google_pilot", b64PubKey, 24*time.Hour)
+	if err != nil {
+		t.Fatalf("Unable to obtain Log metadata: %s", err)
+	}
+
+	// Create signature verifier.
+	sv, err := ct.NewSignatureVerifier(l.PublicKey)
+	if err != nil {
+		t.Fatalf("Couldn't create signature verifier: %s", err)
+	}
+
+	tests := []struct {
+		desc         string
+		sth          *ct.GetSTHResponse
+		receivedAt   time.Time
+		wantErrTypes []reflect.Type
+	}{
+		{
+			desc:       "valid",
+			sth:        validSTH,
+			receivedAt: validReceiveTime,
+		},
+		{
+			desc: "invalid signature",
+			// STH with TreeSize modified (set to 0) so that signature will not
+			// verify.
+			sth: &ct.GetSTHResponse{
+				Timestamp:         validSTH.Timestamp,
+				SHA256RootHash:    validSTH.SHA256RootHash,
+				TreeHeadSignature: validSTH.TreeHeadSignature,
+			},
+			receivedAt: validReceiveTime,
+			wantErrTypes: []reflect.Type{
+				reflect.TypeOf(&SignatureVerificationError{}),
+			},
+		},
+		{
+			desc:       "old STH",
+			sth:        validSTH,
+			receivedAt: invalidReceiveTime,
+			wantErrTypes: []reflect.Type{
+				reflect.TypeOf(&OldTimestampError{}),
+			},
+		},
+		{
+			desc: "invalid sig and old STH",
+			sth: &ct.GetSTHResponse{
+				Timestamp:         validSTH.Timestamp,
+				SHA256RootHash:    validSTH.SHA256RootHash,
+				TreeHeadSignature: validSTH.TreeHeadSignature,
+			},
+			receivedAt: invalidReceiveTime,
+			wantErrTypes: []reflect.Type{
+				reflect.TypeOf(&SignatureVerificationError{}),
+				reflect.TypeOf(&OldTimestampError{}),
+			},
+		},
+	}
+
+	for _, test := range tests {
+		t.Run(test.desc, func(t *testing.T) {
+			sth, err := test.sth.ToSignedTreeHead()
+			if err != nil {
+				t.Fatalf("bad bad: %s", err)
```

Comment:
Could probably have a more helpful error message here ;)

Reply:
Ahahaha oops.  The other error message was "more bad bad" for a while, I fixed that one but clearly forgot about this one. Done.